apps/prairielearn/src/ee/pages/instructorInstanceAdminLti13/instructorInstanceAdminLti13.sql

```diff
@@ -0,0 +1,41 @@
+-- BLOCK select_lti13_instances
+SELECT
+  to_jsonb(lci) AS lti13_course_instance,
+  to_jsonb(li) AS lti13_instance
+FROM
+  lti13_course_instances AS lci
+  JOIN lti13_instances li ON lci.lti13_instance_id = li.id
+WHERE
+  course_instance_id = $course_instance_id
+  AND li.deleted_at IS NULL
+ORDER BY
+  lci.id DESC;
+
+-- BLOCK delete_lti13_course_instance
+WITH
+  old_row AS (
+    DELETE FROM lti13_course_instances AS lci
+    WHERE
+      course_instance_id = $course_instance_id
+      AND id = $lti13_course_instance_id
+    RETURNING
+      lci.*
+  )
+INSERT INTO
+  audit_logs (
+    authn_user_id,
+    table_name,
+    course_instance_id,
+    row_id,
+    action,
+    old_state
+  )
+SELECT
+  $authn_user_id,
+  'lti13_course_instances',
+  old_row.course_instance_id,
+  old_row.id,
+  'delete',
+  to_jsonb(old_row)
+FROM
+  old_row;
```

Comment:
The "modern" way to do a delete + audit log is to start a transaction, then perform the delete and call `insertAuditLog` inside of it. Example:

https://github.com/PrairieLearn/PrairieLearn/blob/2c4da1433ea13ee0996b947a1c946b3839e1881f/apps/prairielearn/src/ee/models/plan-grants.ts#L45

You should add an institution ID and probably a course ID to the log as well; this should be straightforward to do from JS with `res.locals`.

Reply:
Addressed in b2ca77405

Reply:
You missed the "start a transaction" piece; we want the operation and the audit log to be atomic.

Reply:
I mimicked what the plan-grants did from your linked example. I guess it doesn't do that there either? Is there another example?

Reply:
I fixed this up for you in 24e76c7.